### PR TITLE
Android fixes

### PR DESCRIPTION
**Pull request by zavadovsky**
Fix Android client crash and make painting some faster.

### CODE REVIEW

File: client/Android/FreeRDPCore/jni/android_freerdp.c [hunk 1]
```diff
@@ -111,20 +111,44 @@ void android_begin_paint(rdpContext* context)
...
+	if (ninvalid == 0)
+	{
+		DEBUG_ANDROID("ui_update: ninvalid=%d", ninvalid);
+		return;
+	}
+
+	cinvalid = ctx->rdpCtx.gdi->primary->hdc->hwnd->cinvalid;
+
+	x1 = cinvalid[0].x;
+	y1 = cinvalid[0].x + cinvalid[0].y;
```
Comment (akallabeth) [inline client/Android/FreeRDPCore/jni/android_freerdp.c]: @zavadovsky sure that is the correct coordinate?

File: client/Android/FreeRDPCore/jni/android_freerdp.c [hunk 2]
```diff
@@ -111,20 +111,44 @@ void android_begin_paint(rdpContext* context)
...
+	x2 = cinvalid[0].w;
```
Comment (akallabeth) [inline client/Android/FreeRDPCore/jni/android_freerdp.c]: @zavadovsky shouldn't that be ```cinvalid[0].x + cinvalid[0].w```

Comment (zavadovsky) [inline client/Android/FreeRDPCore/jni/android_freerdp.c]: Yes. It's my mistake.
Should be:
x1 = cinvalid[0].x;
y1 = cinvalid[0].y;
x2 = cinvalid[0].x + cinvalid[0].w;
y2 = cinvalid[0].y + cinvalid[0].h;

**Comment by bmiklautz** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
+1

**Comment by bmiklautz** [comment]
+1

**Pull request merged**